#############################################
## Cerate count data for APH & DMSO samples

``` {r}
# load packages
library(ggplot2)
library(ggrepel)
library(plyr)
library(dplyr)
library(Seurat)
library(R.utils)
library(monocle)
library(SingleCellExperiment)
library(reshape2)
# I like the default theme of cowplot
library(cowplot)
library(ggpubr)
library(tidyverse)

# DMSO control sample
filt_data_DMSO= Read10X(data.dir = "~/Analysis_Results/GM_DMSO_SeV_counts/outs/filtered_feature_bc_matrix/")
filt_data_DMSO= CreateSeuratObject(counts= filt_data_DMSO, project = "GM_SeV_DMSO")
dim(filt_data_DMSO) # 54634 11790

# APH sample
filt_data_APH= Read10X(data.dir = "~/Analysis_Results/GM_APH_SeV_counts/outs/filtered_feature_bc_matrix/")
filt_data_APH= CreateSeuratObject(counts= filt_data_APH, project = "GM_SeV_APH")
dim(filt_data_APH) # 54634 12015

#label cell barcodes based on sample name
filt_data_DMSO=RenameCells(filt_data_DMSO, add.cell.id = "DMSO")
filt_data_APH=RenameCells(filt_data_APH, add.cell.id = "APH")

# merge both samples into 1 Seurat object
mat_combined <- merge(filt_data_DMSO, filt_data_APH, add.cell.id=NULL)

# dimension of the combined Seurat object
dim(mat_combined) #54634 23805

# save all the SeV genes in a list
viral_genes = c("N", "P", "M", "F", "HN", "L")

# subset the seurat object with only the viral gene' expressions
ind= subset(mat_combined, features=viral_genes)
head(ind[[]])

# find the cells which have at least 1 copy of at least 1 of the viral mRNAs
inf_seurat= ind[,ind@meta.data$nCount_RNA>0]

dim(inf_seurat) # 6 12812

# save the infection status as a new column in metadata
tmp = ifelse(colnames(mat_combined) %in% colnames(inf_seurat), "Inf", "Not_inf")
mat_combined[["inf_status"]] <- tmp

head(mat_combined)
     
`A data.frame: 10 Ã— 4
orig.ident	nCount_RNA	nFeature_RNA	inf_status
<chr>	<dbl>	<int>	<chr>
DMSO_AAACCCAAGCTGTTAC-1	GM_SeV_DMSO	14908	4566	Inf
DMSO_AAACCCAAGGAACGAA-1	GM_SeV_DMSO	22216	5414	Inf
DMSO_AAACCCACAATGTCTG-1	GM_SeV_DMSO	21204	5256	Inf
DMSO_AAACCCAGTACAAGTA-1	GM_SeV_DMSO	8923	3239	Not_inf
DMSO_AAACCCAGTCCAATCA-1	GM_SeV_DMSO	8392	3223	Not_inf
DMSO_AAACCCAGTCGCCTAG-1	GM_SeV_DMSO	14521	4520	Not_inf
DMSO_AAACCCAGTTAAGTCC-1	GM_SeV_DMSO	19448	5385	Inf
DMSO_AAACCCAGTTCAGCTA-1	GM_SeV_DMSO	15234	4614	Inf
DMSO_AAACCCATCAATCTTC-1	GM_SeV_DMSO	9385	3248	Inf
DMSO_AAACCCATCCACGTCT-1	GM_SeV_DMSO	11829	4171	Not_inf`

# save unfiltered seurat object
saveRDS(mat_combined, file="~/Analysis_Results/R_objs/unfil_mat_combined.rds")

################################################################################
######### Quality control #########

#### no need to remove mt reads because:
# There are not mitochondrial genes aligned to the reads, due to "cellranger 
# mkgtf input.gtf output.gtf --attribute=key:allowable_value" command. 

##### since we have 2 runs combined the cells have more RNA so increase the upper limit from 2500 to 7500
sub_seurat <- subset(mat_combined, subset = nFeature_RNA > 200 & nFeature_RNA < 7500)
dim(sub_seurat) #54634 23284

# save the object for future use
save(sub_seurat, file="~/scRNA-seq/in-house/Robjs/sub_seurat.RData")

# a function for cell percentages across clusters:

stackedbarplot = function(meta, groupx, groupfill, fn, horizontal = F, bold = F, wd = 10, hg = 10, adall = T){

require(tidyverse)
require(dplyr)
require(ggplot2)

clkeys = meta %>% group_by(meta[, c(groupx, groupfill)]) %>% group_keys %>% as.data.frame
clkeys$size = meta %>% group_by(meta[, c(groupx, groupfill)]) %>% group_size 

# Add all cell numbers
if(adall == T){
	spkeys = meta %>% group_by(meta[,groupfill]) %>% group_keys %>% as.data.frame
	colnames(spkeys) = groupfill
	spkeys$size = meta %>% group_by(meta[, groupfill]) %>% group_size
	spkeys = spkeys %>% add_column(tmp = 'AllCells')
	colnames(spkeys)[3] = groupx
	clkeys = rbind(clkeys, spkeys)
}

colnames(clkeys) = c('cluster', 'variable', 'value')

plt = ggplot(clkeys, aes(x = cluster, y = value, fill = variable)) + 
	    geom_bar(position = "fill", stat = "identity") +
	    xlab("") +
	    ylab("Percentage") +
	    theme_classic() +
	    #scale_fill_manual(values = colors) +
	    scale_y_continuous(labels = scales::percent_format()) +
	    theme(text=element_text(size=30), axis.text.y = element_text(face = 'bold')) +
	    rotate_x_text(45) + coord_flip()

if(horizontal == T){
plt = ggplot(clkeys, aes(x = cluster, y = value, fill = variable)) + 
	    geom_bar(position = "fill", stat = "identity") +
	    xlab("") +
	    ylab("Percentage") +
	    theme_classic() +
	    #scale_fill_manual(values = colors) +
	    scale_y_continuous(labels = scales::percent_format()) +
	    theme(text=element_text(size=30)) +
	    rotate_x_text(45)
}

pdf(paste0(fn, '.pdf'), width = wd, height = hg)
print(plt)
dev.off()

	return(plt)

}

# get the stacked bar plots
stackedbarplot(sub_seurat@meta.data, groupx = 'orig.ident', groupfill = 'inf_status', fn="infection_status_breakdown")

meta = sub_seurat@meta.data
ggboxplot(meta, x = 'orig.ident', y = 'infection_score')
ggboxplot(meta, x = 'orig.ident', y = 'infection_score', ylim = c(0,0.1))
ggboxplot(meta, x = 'orig.ident', y = 'infection_score', ylim = c(0,0.05))
ggboxplot(meta, x = 'orig.ident', y = 'infection_score', ylim = c(0,0.1))

```


-Need to remove clusters due to low or no change in the level of marker genes. This means these clusters include no cells (empty droplets). 


```{r}
# packages
library(Seurat)
library(ggplot2)
library(sctransform)
library(harmony)
library(dplyr)
library(cowplot)
library(ggpubr)
library(tidyverse)

# load the seurat object
unfil_mat_combined<- readRDS("~/scRNA-seq/in-house/Robjs/unfil_mat_combined.rds")

dim(unfil_mat_combined) #[1] 54634 23805

# filter the cells for quality control purposes
# draw the graphs
pdf("~/scRNA-seq/in-house/files/preFilter_quality_control_plots.pdf")
VlnPlot(unfil_mat_combined, features = c("nFeature_RNA", "nCount_RNA"), ncol = 2)
dev.off()

# since we have 2 runs combined the cells have more RNA so increase the upper limit from 2500 to 7500
sub_seurat_all <- subset(unfil_mat_combined, subset = nFeature_RNA > 200 & nFeature_RNA < 7500)
dim(sub_seurat_all) #[1] 54634 23284

# normalize and scale the counts

sub_seurat_all <- SCTransform(sub_seurat_all)

# These are now standard steps in the Seurat workflow for visualization and clustering
sub_seurat_all <- RunPCA(sub_seurat_all, verbose = FALSE)
sub_seurat_all <- RunUMAP(sub_seurat_all, dims = 1:30, verbose = FALSE)

sub_seurat_all <- FindNeighbors(sub_seurat_all, dims = 1:30, verbose = FALSE)
sub_seurat_all <- FindClusters(sub_seurat_all, verbose = FALSE)

# save UMAP map pre harmony
pdf("~/scRNA-seq/in-house/files/preHarmony_UMAP.pdf")
DimPlot(sub_seurat_all)
dev.off()

# check IFNB1 expression across seurat clusters
pdf("~/scRNA-seq/in-house/files/preFilter_preHarmony_IFNB1_on_UMAP.pdf")
FeaturePlot(sub_seurat_all, features = "IFNB1", sort.cell = T)
dev.off()

# check for empty droplets
pdf("~/scRNA-seq/in-house/files/preFilter_preHarmony_nCountRNA_boxPlot.pdf")
ggboxplot(sub_seurat_all[[]], x = 'seurat_clusters', y = 'nCount_RNA')
dev.off()

# check the compostion of DMSO and APH samples within seurat clusters
stackedbarplot(sub_seurat_all[[]], groupx='seurat_clusters', groupfill='orig.ident', fn = '~/scRNA-seq/in-house/files/preFilter_preHarmony_composition_of_samples_boxPlot')


# save
save(sub_seurat_all, file= "~/scRNA-seq/in-house/Robjs/sub_seurat_all.RData")

# Remove seurat clusters: 6,7,12, and 13
# as they have low number of gene counts
# load
load(file= "~/scRNA-seq/in-house/Robjs/sub_seurat_all.RData")

# remove clusters and save the R object
table(sub_seurat_all$seurat_clusters)

"   0    1    2    3    4    5    6    7    8    9   10   11 
2505 2275 2123 2105 1973 1696 1623 1565 1557 1426 1246  954 
  12   13   14   15 
 839  604  541  252 "

sub_seurat <- subset(sub_seurat_all, idents = c(0, 1:5, 8:11, 14, 15))
table(sub_seurat$seurat_clusters)
"  0    1    2    3    4    5    6    7    8    9 
2505 2275 2123 2105 1973 1696    0    0 1557 1426 
  10   11   12   13   14   15 
1246  954    0    0  541  252 "

#re-do normalization and scaling
sub_seurat <- SCTransform(sub_seurat)

sub_seurat <- RunPCA(sub_seurat)
sub_seurat <- RunUMAP(sub_seurat, dims = 1:30)

sub_seurat <- FindNeighbors(sub_seurat, dims = 1:30)
sub_seurat <- FindClusters(sub_seurat)

dim(sub_seurat) #[1] 31311 18653

# check if samples are grouped together (DMSO and APH samples)
pdf("~/scRNA-seq/in-house/files/preHarmony_UMAP.pdf")
DimPlot(sub_seurat, group.by= 'orig.ident', raster = TRUE)  
dev.off()

#### this variability can be due to non-biological reasons
# as different indexes were used for each of these samples
# need to remove variability using harmony
sub_seurat <- sub_seurat %>%
  RunHarmony("orig.ident", assay.use = "SCT", lot_convergence = TRUE)

harmony_embeddings <- Embeddings(sub_seurat, 'harmony')
dim(harmony_embeddings) #[1] 18653    50
harmony_embeddings[1:5, 1:5]
"                          harmony_1 harmony_2
DMSO_AAACCCAAGCTGTTAC-1  19.5394159 40.781451
DMSO_AAACCCAAGGAACGAA-1  -0.2868795 31.727112
DMSO_AAACCCACAATGTCTG-1 -10.8458481 -8.090688
DMSO_AAACCCAGTACAAGTA-1  -9.5930742 -7.486736
DMSO_AAACCCAGTCCAATCA-1 -10.1745433 -8.352415
                         harmony_3   harmony_4
DMSO_AAACCCAAGCTGTTAC-1 -12.150360   0.5526155
DMSO_AAACCCAAGGAACGAA-1 -14.154397  17.1421362
DMSO_AAACCCACAATGTCTG-1   7.247603  -3.2248774
DMSO_AAACCCAGTACAAGTA-1  -7.960604  -7.8964792
DMSO_AAACCCAGTCCAATCA-1  14.199246 -11.3519715
                        harmony_5
DMSO_AAACCCAAGCTGTTAC-1 12.925775
DMSO_AAACCCAAGGAACGAA-1 24.987905
DMSO_AAACCCACAATGTCTG-1  3.487940
DMSO_AAACCCAGTACAAGTA-1  1.948819
DMSO_AAACCCAGTCCAATCA-1  5.204003"

#Let's make sure that the datasets are well integrated in the first 2 dimensions after Harmony.
p1 <- DimPlot(object = sub_seurat, reduction = "harmony", pt.size = .1, group.by = "orig.ident")

p2 <- VlnPlot(object = sub_seurat, features = "harmony_1", group.by = "orig.ident", pt.size = .1)

plot_grid(p1,p2)


###########################################################################################

#Many downstream analyses are performed on low dimensional embeddings, not gene expression. 
#To use the corrected Harmony embeddings rather than PCs, set reduction = 'harmony'. 
#For example, let's perform the UMAP and Nearest Neighbor analyses using the Harmony embeddings.

sub_seurat <- sub_seurat %>% 
    RunUMAP(reduction = "harmony", dims = 1:20) %>% 
    FindNeighbors(reduction = "harmony", dims = 1:20) %>% 
    FindClusters(resolution = 0.5) %>% 
    identity()


stackedbarplot(sub_seurat[[]], groupx='seurat_clusters', groupfill='orig.ident', fn = '~/scRNA-seq/in-house/files/postHarmony_composition_of_samples_boxPlot')


stackedbarplot(sub_seurat[[]], groupx='old.ident', groupfill='inf_status', fn = '~/scRNA-seq/in-house/files/postHarmony_composition_of_inf_status_boxPlot')


library(ggrastr)
pdf(file="~/scRNA-seq/in-house/files/postHarmony_seuratClusters.pdf")
DimPlot(sub_seurat, label =T, group.by = 'old.ident', raster = T)
dev.off()

pdf(file="~/scRNA-seq/in-house/files/postHarmony_IFNB1_on_seuratClusters.pdf")
FeaturePlot(sub_seurat, features = "IFNB1", sort= T, reduction = 'umap')
dev.off()

pdf(file="~/scRNA-seq/in-house/files/postHarmony_APHvsDMSO.pdf")
DimPlot(sub_seurat, group.by = 'orig.ident', reduction = 'umap', raster = TRUE)
dev.off()

pdf(file="~/scRNA-seq/in-house/files/postHarmony_inf_status.pdf")
DimPlot(sub_seurat, group.by = 'inf_status', reduction = 'umap',  cols =  c("green", "blue"), raster= T)
dev.off()


#check the dimensions
# data with some clusters removed
dim(sub_seurat) #[1] 31311 18653
# data with all clusters 
dim(sub_seurat_all) # [1]  31739 23284


# check the expression of all IFN genes
allgns = rownames(sub_seurat)
IFN_genes <- allgns[grepl("IFN", allgns)]
(IFN_genes <- IFN_genes[!grepl("R", IFN_genes)])

# create interferon expression status in the metadata
# take the IFN- and + cells within the object
ind= subset(sub_seurat, features= IFN_genes)

# find the cells which have at least 1 copy of at least 1 of the transcript of interest
IFN_seurat= ind[,ind@meta.data$nCount_RNA>0]
head(IFN_seurat)
dim(IFN_seurat) #[1]   24 genes 3258 cells 


# save the IFN status as a new column in metadata
tmp = ifelse(colnames(sub_seurat) %in% colnames(IFN_seurat), "IFN_pos", "IFN_neg")
sub_seurat[["IFN_status"]] <- tmp    

head(sub_seurat)


pdf(file="~/scRNA-seq/in-house/files/postHarmony_IFN_status.pdf")
DimPlot(sub_seurat, group.by = 'IFN_status', reduction = 'umap',  cols = c("purple", "cyan"), raster= T)
dev.off()

# check IFN gene expression across clusters in dot plot
DotPlot(sub_seurat, features = IFN_genes)+rotate_x_text(90)
table(sub_seurat$old.ident) %>% prop.table
"          0           1           2           3           4 
0.149091299 0.141317750 0.127915081 0.120945692 0.102396397 
          5           6           7           8           9 
0.095641452 0.082238782 0.062563663 0.059078969 0.051466252 
         10 
0.007344663 "
sub1 = subset(sub_seurat, subset = old.ident %in% c(3,5,6,7,9,10))
DotPlot(sub1, features = IFN_genes)+rotate_x_text(90)

# perform cell cycle scoring
# A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat.  We can
# segregate this list into markers of G2/M phase and markers of S phase
s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
sub_seurat <- CellCycleScoring(sub_seurat, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

save(sub_seurat, file= "~/scRNA-seq/in-house/Robjs/sub_seurat.RData")
```


```{r}
library(Seurat)

# data
load("~/scRNA-seq/in-house/Robjs/sub_seurat.RData")


# from the UMAPs it looks like the cells are grouped as follows: 
# Clusters 0,2,4, and 8: low percentage of Infected cells with low APH cell percentage
# Clusters 3 and 9: low percentage of Infected cells with high APH cell percentage
# Clusters 5 and 6: high  percentage of Infected cells with low APH cell percentage
# Clusters 7 and 10: high percentage of Infected cells with low APH cell percentage
# So, group the cells in these custom groups
table(sub_seurat$old.ident)
"   0    1    2    3    4    5    6    7    8    9   10 
2781 2636 2386 2256 1910 1784 1534 1167 1102  960  137 "

# set Idents as clusters
Idents(sub_seurat) <- sub_seurat$old.ident
gr1 <- subset(sub_seurat, subset = old.ident %in% c("0", "1", "2", "4", "8"))
dim(gr1) #[1] 31311 10815 it's correct=2781+2636+2386+1910+1102

gr2 <- subset(sub_seurat, subset = old.ident %in% c("3",  "9"))
dim(gr2) #[1] 31311  3216 = 2256+960

gr3 <- subset(sub_seurat, subset = old.ident %in% c("5",  "6"))
dim(gr3) #[1] 31311  3318 = 1784+1534

gr4 <- subset(sub_seurat, subset = old.ident %in% c("7",  "10"))
dim(gr4) #[1] 31311  1304 = 1167+137

# create a new meta column which indicates these custom groups
tmp <- ifelse(colnames(sub_seurat) %in% colnames(gr1), "gr1", ifelse(colnames(sub_seurat) %in% colnames(gr2), "gr2", ifelse(colnames(sub_seurat) %in% colnames(gr3), "gr3", "gr4")))

table(tmp)
#  gr1   gr2   gr3   gr4 
#10815  3216  3318  1304 

sub_seurat$custom_gr <- tmp

library(ggrastr)
pdf(file="~/scRNA-seq/in-house/files/postHarmony_CustomGroups.pdf")
DimPlot(sub_seurat, label =T, raster = T)
dev.off()

############################################################
## set the custom group metadata column as the ident 
Idents(sub_seurat) <- sub_seurat$custom_gr


### perform stacked bar plots of these custom groups
## Infected cell composition
stackedbarplot(sub_seurat[[]], groupx='custom_gr', groupfill='inf_status', fn = '~/scRNA-seq/in-house/files/stackeBbarPlot_custom_gr_infected_cell_composition')

## DMSO/APH composition
stackedbarplot(sub_seurat[[]], groupx='custom_gr', groupfill='orig.ident', fn = '~/scRNA-seq/in-house/files/stackeBbarPlot_custom_gr_APH_DMSO_composition')

# save the seurat object
save(sub_seurat, '~/project/Gozde_data/scRNA-seq/in-house/Robjs/sub_seurat.RData')
###############################################################
```
